app/frontend/Shared/ContextMenu.ts

```diff
@@ -233,52 +229,62 @@ export async function copyLink(context: ContextInfo, win: any) {
   */
 }
 
-export function saveLinkAs(context: ContextInfo, win: any) {
+function saveLinkAs(context: ContextInfo, win: any) {
   download(context.linkURL, win, true, context.suggestedFilename);
 }
 
-export async function copyImage(context: ContextInfo, win: any) {
+async function copyImage(context: ContextInfo, win: any) {
   win = await appGlobal.remoteApp.getWebContents(win.getWebContentsId());
   win.copyImageAt(context.x, context.y);
 }
 
-export async function copyImageURL(context: ContextInfo, win: any) {
+async function copyImageURL(context: ContextInfo, win: any) {
   await copyToClipboard(context.srcURL);
 }
 
-export async function copyVideoURL(context: ContextInfo, win: any) {
+async function copyVideoURL(context: ContextInfo, win: any) {
   await copyToClipboard(context.srcURL);
 }
 
-export function learnSpelling(context: ContextInfo, win: any) {
+function learnSpelling(context: ContextInfo, win: any) {
   win.session.addWordToSpellCheckerDictionary(context.misspelledWord);
 }
 
-export function inspect(context: ContextInfo, win: any) {
+function inspect(context: ContextInfo, win: any) {
   win.inspectElement(context.x, context.y);
   if (win.isDevToolsOpened()) {
     win.devToolsWebContents.focus();
   }
 }
 
-export async function copyToClipboard(text: string | Object) {
+async function copyToClipboard(text: string | Object) {
   if (typeof (text) == "string") {
     await appGlobal.remoteApp.writeTextToClipboard(text);
   } else {
     throw new NotImplemented();
   }
 }
 
-export async function openBrowser(url: URLString) {
+async function openBrowser(url: URLString) {
   await appGlobal.remoteApp.openExternalURL(url);
 }
 
-export function download(url: URLString, window: any, howSaveAsDialog: boolean, filename?: string) {
+async function download(url: URLString, win: any, howSaveAsDialog: boolean, filename?: string) {
   let a = document.createElement("a");
-  a.href = url;
+  let urlObj = new URL(url);
+  if (urlObj.protocol == "http:" || urlObj.protocol == "https:") {
+    let blob = await (await fetch(url)).blob();
+    a.href = await blobToDataURL(blob);
+    filename = filename ?? urlObj.pathname.split("/").pop();
+  } else {
+    a.href = url;
+  }
   a.setAttribute("target", "_blank"); // for non-same-origin images
   if (filename) {
+    sanitize.filename(filename);
     a.setAttribute("download", filename); // TODO sanitize filename
+  } else {
+    a.setAttribute("download", "");
   }
```

Comment:
```suggestion
  a.setAttribute("download", sanitize.filename(filename, ""));
```

Last parameter to sanitize.foo() is a default value, in case the requirement is not met. You can use this to avoid the `if` here entirely.